vendor/runner/cli.js

```diff
@@ -0,0 +1,101 @@
+// Usage:
+//
+// Run all vendors:
+//   npm run test:vendor
+//
+// Run some vendors:
+//   npm run test:vendor -- --only=vendor1,vendor2,vendor3
+//
+// Infer the current PR from the Travis environment, and look for bracketed,
+// space-separated vendor names in the pull request title. If none are found,
+// do not run any tests. For example:
+// Pull request title: [travis sonar] Support user token authentication
+//   npm run test:vendor -- --pr
+// is equivalent to
+//   npm run test:vendor -- --only=travis,sonar
+
+'use strict';
+
+const request = require('request');
+const minimist = require('minimist');
+const difference = require('lodash.difference');
+const serverHelpers = require('../../test/in-process-server-helpers');
+const Runner = require('./runner');
+
+const getTitle = (repoSlug, pullRequest) => new Promise((resolve, reject) => {
+  const options = {
+    uri: `https://api.github.com/repos/${repoSlug}/pulls/${pullRequest}`,
+    json: true,
+    headers: { 'User-Agent': 'badges/shields' },
+  };
+  request(options, (err, res, json) => {
```

Comment:
One option is to just create a simple separate function that wraps `request`, something like:
```
function requestPromise(options) {
  return new Promise((resolve, reject) => {
    request(options, (err, res, json) => {
      if (err) {
        reject(err);
      } else {
        resolve({json, res});
      }
    });
  });
}
```

> If I use node-fetch I'm adding a whole interface that devs wouldn't be used to.

`fetch` is standardised now (and I think modern browsers even have it available out-of-the-box), so hopefully more and more developers are getting used to it :)

> If I use request and the person doesn't understand promises they'll be confused

Promises are becoming more and more common though. Pretty much anyone dealing with a modern JS codebase will see them :D  Sure, async/await is coming, but that just builds on top of promises.